packages/react-vapor/src/components/textInput/useForm.ts

```diff
@@ -0,0 +1,12 @@
+import {useContext, useMemo} from 'react';
+import {TextInputState} from './TextInputReducer';
+import {FormComponent, FormContext} from '../form/FormProvider';
+
+export const useForm = (form: FormComponent): {state: Record<string, TextInputState>} => {
```

Comment:
I feel the name isn't quite right. We don't use the whole form here. A form typically consists of multiple form components (for example a text input, some checkboxes and a single select). But here the purpose of the hook is to probe and/or interact with multiple instances of the same form component type.

```suggestion
export const useFormComponent = (component: FormComponent): {state: Record<string, TextInputState>} => {
```